pkg/api/v1alpha1/snowdatacenterconfig_types.go

```diff
@@ -44,6 +57,22 @@ func (s *SnowDatacenterConfig) ClearPauseAnnotation() {
 	}
 }
 
+func (s *SnowDatacenterConfig) SetDefaults() {
+	if s.Spec.IdentityRef == nil {
+		s.Spec.IdentityRef = &Ref{
+			Kind: SnowIdentityKind,
+			Name: fmt.Sprintf("%s-snow-credentials", s.GetName()),
+		}
+	}
+}
+
+func (s *SnowDatacenterConfig) Validate() error {
+	if s.Spec.IdentityRef != nil && s.Spec.IdentityRef.Kind != SnowIdentityKind {
```

Comment:
Ahhh this is interesting
We are allowing nil `IdentityRef` in this validation. Which makes sense for the CLI (assuming we run this before setting defaults, which maybe we shouldn't).

But it doesn't work for the webhook. So we probably need to add a second "layer" of validations that are run after or the defaults.

Ideally, all defaults, including the ones in the api package, the ones in the `cluster` package and the ones in the `snow` provider package would run before we do any validations. However, that will require some refactoring to the CLI flow. That's why I think the second layer of validations might be the best solution here, as long as we document the methods well.